File: src/bigo/AddUpTo1.js
/*
 Suppose we want to write a function that calculates the sum of all numbers 
 from 1 up to (and including) some number n.
*/

function addUpTo(n) {
  let total = 0
  for (let i = 0; i <= n; i++) {
    total += i;
  }
  return total
}

const total = addUpTo(3);
console.log('Total: ', total)

File: src/bigo/AddUpTo1.java
package bigo;

/**
 * Suppose we want to write a function that calculates the sum of all numbers
 * from 1 up to (and including) some number n.
 */
class AddUpTo1 {
  public static void main(String[] args) {

    long number = 3;

    long total = addUpTo(number);
    System.out.println("Total: " + total);
  }

  /**
   * First: naive approach O(n) complexity
   * @param n
   * @return
   */
  public static long addUpTo(long n) {
    long total = 0;
    for (int i = 1; i <= n; i++) {
      total += i;
    }
    return total;
  }
}